ssz-rs/src/lib.rs

```diff
@@ -13,18 +18,42 @@ mod uint;
 mod union;
 mod vector;
 
-use crate::list::Error as ListError;
-use crate::vector::Error as VectorError;
+use crate::lib::*;
```

Comment:
I admit this is nitpicky, but...

I don't love tossing all of the random symbols from `crate::lib::*;`  into the `lib` module namespace

but it looks like we did this for the fns `serialize` and `deserialize`, along with the `SimpleSerializeError`

to resolve this, I'd suggest making a new file `utils.rs` with `serialize and `deserialize` (and the required imports)

along with moving the error type to a new file `error.rs` (and while we are at it just calling it `Error`)

Reply:
ok, I made these changes on `main`, if you rebase you'll see what needs to happen (should just be a few `crate::lib::*` imports)

Reply:
Much cleaner, thanks. I adjusted my changes accordingly.